web/src/components/Broadcast.js

```diff
@@ -0,0 +1,52 @@
+import { useEffect } from 'react';
```

Comment:
It seems a little weird to put this into a React component, but it gives us two advantages that I can see:
1. It makes initialization explicit, rather than happening when the module loads, which makes it easier to test. We could accomplish this with an `init` method or something similar as well, but...
2. It allows us to hook when the DOM begins to unload so we can also close the broadcast channel. I'm not sure if it's necessary in the case of the page shutting down, but because it's a shared resource across tabs, I was worried about memory leaks and the browser never being able to release the channel once created.

Reply:
MAGIC